source/main/cardano/CardanoNode.js

```diff
@@ -340,84 +343,125 @@ export class CardanoNode {
-
-        node
-          .start()
-          .then((api) => {
-            const processes: {
-              wallet: ChildProcess,
-              node: ChildProcess,
-            } = {
-              wallet: node.walletService.getProcess(),
-              node: node.nodeService.getProcess(),
-            };
-
-            // Setup event handling
-            node.walletBackend.events.on('exit', (exitStatus) => {
-              _log.info('CardanoNode#exit', { exitStatus });
-              const { code, signal } = exitStatus.wallet;
-              this._handleCardanoNodeExit(code, signal);
-            });
+      if (isSelfnode) {
+        try {
+          const { selfnodeBin, mockTokenMetadataServerBin } = launcherConfig;
+          const { node, replyPort } = await CardanoSelfnodeLauncher({
+            selfnodeBin,
+            mockTokenMetadataServerBin,
+            processName: CARDANO_PROCESS_NAME,
+            onStop: this._ensureProcessIsNotRunning,
+          });
+          _log.info(
+            `CardanoNode#start: cardano-node child process spawned with PID ${node.pid}`,
+            { pid: node.pid }
+          );
+          this._node = node;
+          this._handleCardanoNodeMessage({ ReplyPort: replyPort });
+          resolve();
+        } catch (error) {
+          _log.error(
+            'CardanoNode#start: Unable to initialize cardano-launcher',
+            { error }
+          );
+          const { code, signal } = error || {};
+          this._handleCardanoNodeError(code, signal);
+          reject(
+            new Error(
+              'CardanoNode#start: Unable to initialize cardano-launcher'
+            )
+          );
+        }
+      } else {
+        try {
+          const node = await CardanoWalletLauncher({
+            nodeImplementation,
+            nodeConfig,
+            cluster,
+            stateDir,
+            tlsPath,
+            block0Path,
+            block0Hash,
+            secretPath,
+            configPath,
+            syncTolerance,
+            nodeLogFile,
+            walletLogFile,
+            cliBin,
+            isStaging,
+            metadataUrl,
+          });
 
-            node.pid = processes.node.pid;
-            node.wpid = processes.wallet.pid;
-            node.connected = true; // TODO: use processes.wallet.connected here
-            _log.info(
-              `CardanoNode#start: cardano-node child process spawned with PID ${processes.node.pid}`,
-              { pid: processes.node.pid }
-            );
-            _log.info(
-              `CardanoNode#start: cardano-wallet child process spawned with PID ${processes.wallet.pid}`,
-              { pid: processes.wallet.pid }
-            );
-            this._handleCardanoNodeMessage({
-              ReplyPort: api.requestParams.port,
+          this._node = node;
+
+          _log.info('Starting cardano-node now...');
+
+          // await promisedCondition(() => node.connected, startupTimeout);
+
```

Comment:
Do we need this for some future cases / checks ?

Reply:
Yes

Reply:
👍